src/lib/api/project/compare-branches.ts

```diff
@@ -0,0 +1,34 @@
+import debug = require("debug");
+import { requestsManager } from "snyk-request-manager";
+import { updateProject } from ".";
+import { CompareAndUpdateBranchesResponse } from "../../types";
+
+export async function compareAndUpdateBranches(
+    requestManager: requestsManager,
+    snykDefaultBranch: string,
+    githubBranch: string,
+    projectId: string,
+    orgId: string,
+  ): Promise<CompareAndUpdateBranchesResponse> {
+
+
+    if (!projectId) {
+        throw new Error(
+          `Missing required parameters. Please ensure you have set: projectId.
+          \nFor more information see: https://snyk.docs.apiary.io/#reference/projects/individual-project/update-a-project`,
+        );
+    }
+
+    if (snykDefaultBranch != githubBranch) {
+
+        debug(`default branch update needed for project ${projectId}\n`);
+        const projectUpdated = await updateProject(requestManager, orgId, projectId, {branch: githubBranch});
+        return { ProjectNeededUpdate: true, projects: projectUpdated}
```

Comment:
I think we can just return `updated: true` or `updated: false` and throw and error if we fail to update